samples/subscribingToStream.go

```diff
@@ -0,0 +1,168 @@
+package samples
+
+import (
+	"context"
+	"time"
+
+	"github.com/EventStore/EventStore-Client-Go/client"
+	"github.com/EventStore/EventStore-Client-Go/client/filtering"
+	"github.com/EventStore/EventStore-Client-Go/options"
+	"github.com/EventStore/EventStore-Client-Go/position"
+	"github.com/EventStore/EventStore-Client-Go/stream_position"
+)
+
+func SubscribeToStream(db *client.Client) {
+	// region subscribe-to-stream
+	opts := options.SubscribeToStreamOptionsDefault()
+
+	stream, err := db.SubscribeToStream(context.Background(), "some-stream", &opts)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer stream.Close()
+
+	for {
+		event := stream.Recv()
+
+		if event.EventAppeared != nil {
+			// handles the event...
+		}
+
+		if event.Dropped != nil {
+			break
+		}
+	}
+	// endregion subscribe-to-stream
+
+	// region subscribe-to-stream-from-position
+	opts = options.SubscribeToStreamOptionsDefault().Position(stream_position.Revision(20))
+
+	db.SubscribeToStream(context.Background(), "some-stream", &opts)
+	// endregion subscribe-to-stream-from-position
+
+	// region subscribe-to-stream-live
+	opts = options.SubscribeToStreamOptionsDefault().Position(stream_position.End())
+
+	db.SubscribeToStream(context.Background(), "some-stream", &opts)
+	// endregion subscribe-to-stream-live
+
+	// region subscribe-to-stream-resolving-linktos
+	opts = options.SubscribeToStreamOptionsDefault().Position(stream_position.Start()).ResolveLinks()
+
+	db.SubscribeToStream(context.Background(), "$et-myEventType", &opts)
+	// endregion subscribe-to-stream-resolving-linktos
+
+	// region subscribe-to-stream-subscription-dropped
+	var offset stream_position.StreamPosition = stream_position.Start()
+	for {
+		var stream *client.Subscription = nil
+
+		if stream == nil {
+			opts := options.SubscribeToStreamOptionsDefault().Position(offset)
+			stream, err := db.SubscribeToStream(context.Background(), "some-stream", &opts)
+
+			if err != nil {
+				 time.Sleep(1 * time.Second)
+			}
+
+			for {
+				event := stream.Recv()
+
+				if event.Dropped != nil {
```

Comment:
I think that it's visible here even better that event.SubscriptionDropped may be a better name :wink: